src/checkers/inference/solver/backend/AbstractFormatTranslator.java

```diff
@@ -48,9 +51,9 @@ public AbstractFormatTranslator(Lattice lattice, ConstraintVerifier verifier) {
         this.verifier = verifier;
     }
 
-    /**Remember to call this method at the end of subclass FormatTranslator constructor(MaxSAT and LogiQL) or at the last
+    /** Remember to call this method at the end of subclass FormatTranslator constructor(MaxSAT and LogiQL) or at the last
      * step of initializing FormatTranslator - in Z3BitVector backend, it's initSolver() method. Because the creation of
-     * encoders might not only need the parameters - lattice and verifier, but also need fields depending on concrete backend
+     * encoders might not only need the parameter - verifier, but also need fields depending on concrete backend
      * So after those dependant fields are initialized, call this method to finish initializing encoders*/
     protected void postInit() {
```

Comment:
javadoc usually starts with one or two sentences, explain the purpose/main functionalities of this method/field. Then might have a separate paragraph for detail explanations of programming guide, conventions, etc.

Also, if this method is specific for post initializing encoders, rename it to 'postInitEncoders()`. (And it seems this is just the place initializing encoders? Maybe just call it `initEncoders()` as we don't have another place initializing encoders prior than this place...)

Better to refactor it to something like:

```java

/**
* Initialize concrete encoders that used by this format translator.
*
* It is sub-class responsibility to decide how to construct encoders, and 
* when is a proper timing to initialize encoders.
*
* E.g. For MaxSat and LogiQL, it initialize encoders with additional informations other than verifiers,
* and initialize them at the last step of it's FormatTranslator constructors.
*/
 protected void postInit() {
...
}
```

Reply:
Also one thought: it seems this `postInit` method doesn't get called in this abstract class. This is seems a bad style: define a method implementation in an abstract class, but actually abstract class doesn't use it in any template it defines.

Could we enforce this method get called by the constructor in this abstract class? (or remove this method, just call all `createXxxEncoder()` methods in the constructor.)

For Z3, I can have a `setContext(context)` for each concrete encoder, and just set the context in `Z3FormatTranslator` constructor after the super constructor initialized those encoders.